pkg/gen/nodejs-templates/provider.ts.mustache

```diff
@@ -174,10 +197,16 @@ export namespace yaml {
     export class ConfigFile extends CollectionComponentResource {
         constructor(name: string, config?: ConfigFileOpts, opts?: pulumi.ComponentResourceOptions) {
             super("kubernetes:yaml:ConfigFile", name, config, opts);
-            const text = fs.readFileSync(config && config.file || name).toString();
-            const objs = jsyaml.safeLoadAll(text);
-            this.resources = pulumi.output(parseYamlDocument({
-                objs: objs,
+            const fileId = config && config.file || name
+            let text: Promise<string>;
+            if (fileId.startsWith("http://") || fileId.startsWith("https://")) {
+                text = got(fileId).then(r => r.body)
+            } else {
+                text = new Promise<string>(resolve => resolve(fs.readFileSync(fileId).toString()))
```

Comment:
This is fine - but:
1. You could do the same thing with just `Promise.resolve(fs.readFileSync(fileId).toString())`.
2. You sort of may as well do async file I/O here too at this point